workflow-starfusion/src/main/java/ca/on/oicr/pde/workflows/STARfusionWorkflowClient.java

```diff
@@ -0,0 +1,189 @@
+        file1.setSourcePath(read2Fastq);
+        file1.setType(FASTQ_GZIP_MIMETYPE);
+        file1.setIsInput(true);
+        return this.getFiles();
+    }
+
+    @Override
+    public void buildWorkflow() {
+
+        /**
+         * Steps for sequenza: 1. Check if "bam" file exists; true 2. Check if
+         * "bai" file exists; true: go to step 4 3. Check if normal Pb_R sample
+         * exists; true: go to step 4; else abort 3. If false: samtools index
+         * "bam" file 4. Run job sequenza-utils 5. If outputFile ends with
+         * "bin50.gz"; go to step 6; else go to step 4 6. Run job sequenzaR 7.
+         * Iterate through the files/folders in outDir: 8. If fileName1 ==
+         * "pandc.txt" and fileName2 ends with "Total_CN.seg"; create a folder
+         * called "copynumber" 9. If fileType == "folder"; create a folder
+         * called "model-fit"; move folders to "model-fit" 10. If fileType ==
+         * "file" && fileName != outputFile; move file to "model-fit" 11. Delete
+         * outputFile (rm outputFile) 12. zip "model-fit" 13. outputFile =
+         * fileName2 14. OutputDir contains the following: fileName1,
+         * outputFile, model-fit.zip
+         */
+        // workflow : read inputs read1 fastq and read2 fastq file; run star-fusion; write the output to temp directory; 
+        // run sequenzaR; handle output; provision files (3) -- .tsv, .tsv, .tsv;
+        Job parentJob = null;
+        this.outDir = this.outputFilenamePrefix + "_output";
+
+   
+        Job starJob = runStarFusion();
+        parentJob = starJob;
+      
+
+        // Provision .seg, .varscanSomatic_confints_CP.txt, model-fit.tar.gz files
+        String fusionPredictionTsv = this.tmpDir + "star-fusion.fusion_predictions.tsv";
+        SqwFile fusionTSV = createOutputFile(fusionPredictionTsv, TXT_METATYPE, this.manualOutput);
+        fusionTSV.getAnnotations().put("STAR_fusion_prediction_tsv", "STAR_fusion");
+        starJob.addFile(fusionTSV);
+
+        String fusionAbridgedTsv = this.tmpDir + "star-fusion.fusion_predictions.abridged.tsv";
+        SqwFile abridgedTSV = createOutputFile(fusionAbridgedTsv, TXT_METATYPE, this.manualOutput);
+        abridgedTSV.getAnnotations().put("STAR_fusion_abridged_tsv", "STAR_fusion");
+        starJob.addFile(abridgedTSV);
+        
+        String FFP_coding_effect = this.tmpDir + "star-fusion.fusion_predictions.abridged.coding_effect.tsv";
+        SqwFile codingTSV = createOutputFile(FFP_coding_effect, TXT_METATYPE, this.manualOutput);
+        codingTSV.getAnnotations().put("STAR_fusion_coding_effect_tsv ", "STAR_fusion");
+        starJob.addFile(codingTSV);
+          }
+
+  
+    private Job runStarFusion() {
+        Job starJob = getWorkflow().createBashJob("starfusionjob");
+        Command cmd = starJob.getCommand();
+        cmd.addArgument("export LD_LIBRARY_PATH=/oicr/local/analysis/sw/perl/perl-5.22.2-tgl/lib:$LD_LIBRARY_PATH" +";");
```

Comment:
I see that lot's of software version info is hard-coded, although we don't usually customize software we run with workflows every time, it will be easier to update the next version of STAR-fusion if all of these is parametrized with variables. Variables may be sitting in the default workflow.ini and populated during a build (check out other workflows for an example, we use this method pretty much everywhere)